src/applications/vaos/appointment-list/components/ExpressCareDetailsPage.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect } from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import moment from 'moment';
+import AlertBox from '@department-of-veterans-affairs/component-library/AlertBox';
+
+import {
+  APPOINTMENT_STATUS,
+  UNABLE_TO_REACH_VETERAN_DETCODE,
+} from '../../utils/constants';
+import { sentenceCase } from '../../utils/formatters';
+import { scrollAndFocus } from '../../utils/scrollAndFocus';
+
+import { getPatientTelecom } from '../../services/appointment';
+import { selectExpressCareRequestById } from '../redux/selectors';
+import PageLayout from './AppointmentsPage/PageLayout';
+
+function ExpressCareDetailsPage({ appointment }) {
+  const appointmentDate = moment.parseZone(appointment?.start);
+  const canceled = appointment?.status === APPOINTMENT_STATUS.cancelled;
+
+  useEffect(() => {
```

Comment:
I set this up a little different, because there's no easy way to tell where the already fetch EC request might be in the state just by looking at a status variable. So I put logic to get the right appointment in a selector and am redirecting if the appointment isn't there. Once we start fetching the specific endpoints, it shouldn't be too hard to deal with both paths, but the temporary way the other pages are using doesn't quite work for EC.